extensions/amp-experiment/1.0/amp-experiment.js

```diff
@@ -175,101 +173,6 @@ export class AmpExperiment extends AMP.BaseElement {
-    // As that would take more time, and affect the user,
-    // vs. help the developer.
-    let totalMutations = 0;
-    const experimentToVariantKeys = Object.keys(experimentToVariant);
-
-    for (let i = 0; i < experimentToVariantKeys.length; i++) {
-      const experimentKey = experimentToVariantKeys[i];
-      const variantKey = experimentToVariant[experimentKey];
-      const variant =
-        /** @type {!JsonObject} */ (config[experimentKey]['variants'][
-          variantKey
-        ]);
-      totalMutations += variant['mutations'].length;
-    }
-
-    if (totalMutations > MAX_MUTATIONS) {
-      const numMutationsError =
-        'Max number of mutations for the total ' +
-        `applied experiments exceeded: ${totalMutations} > ` +
-        MAX_MUTATIONS;
-      user().error(TAG, numMutationsError);
-      throw new Error(numMutationsError);
-    }
-  }
-
-  /**
-   * Passes the given experiment and variant pairs to the correct handler,
-   * to apply the experiment to the document.
-   * Experiment with no variant assigned (null) will be skipped.
-   *
-   * For example, the `experimentToVariant` object looks like:
-   * {
-   *   'appliedExperimentName': 'chosenVariantName',
-   *   'anotherAppliedExperimentName': 'chosenVariantName'
-   * }
-   * Which is a simplified version of the config and
-   * represents what variant of each experiment
-   * should be applied.
-   *
-   * @param {!JsonObject} config
-   * @param {!Object<string, ?string>} experimentToVariant
-   * @return {!Promise<!Object<string, ?string>>} a promise of the original
-   *     param passed in
-   * @private
-   */
-  applyExperimentVariants_(config, experimentToVariant) {
-    const appliedExperimentToVariantPromises = [];
-
-    for (const experimentName in experimentToVariant) {
-      const variantName = experimentToVariant[experimentName];
-      if (variantName) {
-        const variantObject = config[experimentName]['variants'][variantName];
-        appliedExperimentToVariantPromises.push(
-          this.applyMutations_(experimentName, variantObject)
-        );
-      }
-    }
-
-    return Promise.all(appliedExperimentToVariantPromises).then(
-      () => experimentToVariant
-    );
-  }
-
-  /**
-   * Passes the given experimentName and variantObject pairs
-   * to the mutation service to be applied to the document.
-   * @param {string} experimentName
-   * @param {!JsonObject} variantObject
-   * @return {!Promise}
-   * @private
-   */
-  applyMutations_(experimentName, variantObject) {
-    const doc = this.getAmpDoc();
-    return doc.whenReady().then(() => {
-      // Parse / Validate all of our mutations
-      const mutationOperations = variantObject['mutations'].map(mutation =>
-        parseMutation(mutation, this.win.document)
-      );
-
-      // Apply our mutations
-      mutationOperations.forEach(mutationOperation => mutationOperation());
-    });
-  }
-
   /**
    * Function to check if recaptcha experiment is enabled,
```

Comment:
oops : )

Reply:
This is intentional 😄 I moved this into `apply-experiment.js` in the larger application function 👍

Reply:
haha, I meant "to check if **recaptcha** experiment" : )